[v6r15] Various fixes

fstagni opened the pull request:
- fixed double creation of db tables in SystemLoggingDB
- added some logging to RSS InfoGetter
- simplification of code for RSS InfoGetter
- made some tests discoverable by py.test
- added/reviving tests for ConsistencyInspector
- Jenkins/CI shell scripts: added some protections for navigations

chaen commented on ResourceStatusSystem/Utilities/InfoGetter.py:
```diff
@@ -1,217 +1,57 @@
-  """
-    Class InfoGetter is in charge of getting information from the RSS Configurations
+
+def sanitizedecisionParams( decisionParams ):
```
why taking it out of the class ? It logically belongs there no ?

fstagni replied:
To be honest I think the whole class is useless here. Just functions would be enough

chaen replied:
Okay but then why doing it only for one method ? It seems more logical to do it for all or none

chaen commented on ResourceStatusSystem/Utilities/InfoGetter.py:
```diff
@@ -270,32 +115,29 @@ def __getPoliciesThatApply( self, decisionParams ):
                      'type' : policyType,
-                     'args' : {}
-                   }
+                     'args' : {} }
```
If we agree to follow pep8 as I think we did, the previous convention is the correct one

chaen replied:
Maybe went unnoticed, so I just poke that one

chaen commented on ResourceStatusSystem/Utilities/InfoGetter.py:
```diff
@@ -355,287 +197,77 @@ def __getPolicyActionsThatApply2( decisionParams, singlePolicyResults,
-      if policyMatch:
-        policyActionsThatApply.append( policyActionName )
+    unknownDomains = list( set(targetDomain) - set(domainNames) )
+    if len(unknownDomains) > 0:
```
```if unknownDomains:```

chaen commented on ResourceStatusSystem/Utilities/InfoGetter.py:
```diff
@@ -355,287 +197,77 @@ def __getPolicyActionsThatApply2( decisionParams, singlePolicyResults,
 
-    return S_OK( policyActionsThatApply )
+    knownDomains = list( set(domainNames) & set(targetDomain) )
+    if len(knownDomains) == 0:
```
```if not knowDomains:```

chaen commented on ResourceStatusSystem/Utilities/InfoGetter.py:
```diff
@@ -355,287 +197,77 @@ def __getPolicyActionsThatApply2( decisionParams, singlePolicyResults,
+          else:
+            gLogger.info( "ComputingElement %s found in domains %s" % ( name, domains ) )
+        else:
+          gLogger.warn( "unable to verify if ComputingElement %s is in domains %s" % ( name, domains ) )
```
no return false here ?

chaen commented:
I don't see how commit 0fed8f6152bd5d6e7f892c8e55c987bfea2c5d64 fixes the double creation ? There is still nowhere a check before calling ``_createTables```

fstagni commented:
The tables creation was invoked both in the dB and in the service classes

chaen commented:
I find terribly misleading that ```checkTable``` actually does ```createTable```. 
But are you sure that this is enough ? Next time you will restart the service, it will call ```checkTable``` again, so the problem is the same as calling it twice.

fstagni commented:
Yes, this `checkTables` actually creates the tables when they are not there. In that same mentioned commit I removed the call to it from the service, so now it's only in the DB class. It's tested in certification. BTW I also updated with the suggested fixes.

chaen commented:
Yes you are right, it works, but I don't understand why :-) I'll check the rest

chaen commented on tests/Integration/Framework/Test_LoggingDB.py:
```diff
@@ -9,3 +9,131 @@
 
 
 DIRAC.exit()
```
Unconditional exit, just like that ? :D

chaen commented on tests/Jenkins/dirac_ci.sh:
```diff
@@ -246,7 +246,10 @@ function DIRACPilotInstall(){
 
 	#Don't launch the JobAgent here
+	cwd=$PWD
+	cd $PILOTINSTALLDIR
```
for safety, maybe check that the cd was successful

chaen commented on tests/Jenkins/utilities.sh:
```diff
@@ -43,6 +43,25 @@ function default(){
+		cp $TESTCODE/DIRAC/releases.cfg $TESTCODE/releases.cfg
+	else
+		cwd=$PWD
+		cd $TESTCODE/DIRAC/
```
test the rc

atsareg commented on Resources/Storage/test/test_StorageElement.py:
```diff
@@ -4,8 +4,8 @@
 
```
Should be renamed to Test_StorageElement.py ?

atsareg commented on Resources/Storage/test/test_StorageFactory.py:
```diff
@@ -2,8 +2,8 @@
 
```
ibid

atsareg commented on Resources/Storage/test/test_StoragePlugIn.py:
```diff
@@ -5,8 +5,8 @@
 
```
ibid

atsareg commented:
Review OK

fstagni commented:
I left the "test_" instead of "Test_" for some of the files because the tests in them just don't work as of now, and would just make Travis to fail. They need to be reviewed and changed, or moved.

petricm commented:
Maybe we should distinguish somehow what is a pure python unit test and what needs additional infrastructure like CS, SE, etc. ?

atsareg commented:
I thought that this is what should go to the Jenkins integration and tests in the systems

fstagni commented:
Exactly: everything that needs something else than "pure python" goes to the tests/ directory. So, for example everything that needs a cfg, and that needs to start with `parseCommandLine()` goes there.

fstagni commented:
For the tests in question, they are at the moment some abandonware that should be revisited, and maybe moved. For the moment, I left them with lower-case 't' because this way they won't be discovered and therefore won't be run by Travis.
